qiskit_experiments/framework/backend_timing_mixin.py

```diff
@@ -0,0 +1,169 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+import math
+from typing import Union
+
+from qiskit import QiskitError
+
+
+class BackendTimingMixin:
+    """Mixin class for ``BaseExperiment`` to provide timing helper methods
+
+    The methods and properties provided by this class help with calculating
+    delay and pulse timing that depends on the timing constraints of the
+    backend. They abstract away the necessary accessing of the backend object.
+
+    .. note::
+
+        The methods in this class assume that the ``backend`` attribute is
+        constant. Methods should not call methods of this class before and
+        after modifying the ``backend`` attribute and expect consistent
+        results.
+    """
+
+    @property
+    def _delay_unit(self) -> str:
+        """The delay unit for the current backend
+
+        "dt" is used if dt is present in the backend configuration. Otherwise
+        "s" is used.
+        """
+        if self.backend is not None and hasattr(self.backend.configuration(), "dt"):
+            return "dt"
+
+        return "s"
+
+    @property
+    def _dt(self) -> float:
+        """Backend dt value
+
+        Raises:
+            QiskitError: The backend is not set or does not include a dt value.
+        """
+        if self.backend is not None and hasattr(self.backend.configuration(), "dt"):
+            return self.backend.configuration().dt
+
+        raise QiskitError("Backend must be set to consider sample timing.")
```

Comment:
Perhaps you need another message for the case when a backend exists but dt is not reported.

Reply:
Yes, I can add that, but I will wait until we resolve some other conversations first (like whether there should be an error or just return 1 here).

Reply:
The two error cases are split out now (they occur in different methods).